### PR TITLE
CLC-4776 - Site Provisioning: Update API for course creation to support any site creation

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.ets.berkeley.edu/jira/browse/CLC-4776

### CODE REVIEW

File: app/models/canvas/public_authorizer.rb [hunk 1]
```diff
@@ -6,14 +6,16 @@ def initialize(canvas_user_id)
...
+    def can_create_site?
       self.class.fetch_from_cache @canvas_user_id do
         authorization = false
         campus_uid = Canvas::UserProfile.new(:canvas_user_id => @canvas_user_id).login_id
         if campus_uid
           user = AuthenticationState.new(user_id: campus_uid)
           policy = user.policy
-          authorization = policy.can_create_canvas_course_site?
+          # if you cannot create a project site (i.e. you are not affiliated as a staff or faculty member),
+          # then you surely will not have official sections available to create a course site
```
Comment (paulkerschen) [inline app/models/canvas/public_authorizer.rb]: Is this absolutely always true? Can GSIs create course sites?

File: spec/controllers/canvas_controller_spec.rb [hunk 2]
```diff
@@ -36,29 +36,44 @@
...
-        expect(response_json['canCreateCourseSite']).to be_falsey
+        expect(response_json['canCreateSite']).to eq false
       end
     end
 
     context "when user is authorized to create course site" do
-      before { allow_any_instance_of(Canvas::PublicAuthorizer).to receive(:can_create_course_site?).and_return(true) }
+      before { allow_any_instance_of(Canvas::PublicAuthorizer).to receive(:can_create_site?).and_return(true) }
+
+      it 'includes Access-Control-Allow-Origin header' do
```
Comment (paulkerschen) [inline spec/controllers/canvas_controller_spec.rb]: Should this be a shared example?

File: app/models/canvas/public_authorizer.rb (hunk 1 above)
Comment (redconfetti) [inline app/models/canvas/public_authorizer.rb]: Yes. GSI's can create course sites, and I've confirmed that they do have either a staff or faculty affiliation. For instance, a GSI may be associated with a discussion group section, and can create a course site for this purpose. @jonmhays, can you confirm this?

Comment (jonmhays) [inline app/models/canvas/public_authorizer.rb]: @redconfetti & @paulkerschen, yes GSIs are often assigned as an instructor for a secondary sections.  In these cases they can create course sites.  They are also treated as academic staff and can therefore create project sites.

**Comment by paulkerschen** [comment]
Half happy Travis.

**Pull request merged**